config-ui/src/hooks/useDataScopesManager.jsx

```diff
@@ -137,7 +138,8 @@ function useDataScopesManager ({ provider, blueprint, /* connection, */ settings
           newScope = projects[connection.id]?.map((p) => ({
             ...newScope,
             options: {
-              projectId: Number(p),
+              projectId: Number(p.value),
```

Comment:
Use optional chaining check here (`?`)
```suggestion
              projectId: Number(p?.value),
```

Reply:
We shouldn't use `?.` too much if we could know p is not null. Because it will cover up some bugs such as p is not a object.

Reply:
It's not about covering up bugs, for vars that are dynamic when components are unmounted/mounted react will throw a warning unnecessarily. Optional chaining will allow code-flow to continue even though we don't expect or prefer p to be null. Any valid bug will present itself in the UI.

Reply:
Here p is from projects. projects maybe is an empty array. But it won't be a null array.

Reply:
Leave chaining off for now and we'll see if React throws an error in dev or built mode.

Reply:
No error in console after add blueprint and edit blueprint except this. Because of #2994, I cannot go to other page.

![image](https://user-images.githubusercontent.com/3294100/188793726-d634b4b0-6b75-4a6b-a9b3-e2dbd57f9e86.png)

Reply:
#2994 looks to be a new issue, I still need to local test this PR.